hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestStreamingSource.scala

```diff
@@ -48,159 +48,176 @@ class TestStreamingSource extends StreamTest {
   }
 
   test("test cow stream source") {
-    withTempDir { inputDir =>
-      val tablePath = s"${inputDir.getCanonicalPath}/test_cow_stream"
-      HoodieTableMetaClient.withPropertyBuilder()
```

Comment:
Can we just override the whole class and make the flag `DataSourceReadOptions.INCREMENTAL_FETCH_INSTANT_BY_STATE_TRANSITION_TIME` overridable?